### DISCUSSION TITLE
AI-generated hashtags

### DISCUSSION THREAD

**Discussion by chetanyakan** [mattermost/community#100] [Moonshot Ideas]
Use a plugin to auto mark certain posts with AI-generated hashtags which can be hidden in the UI by default.
Users can then search for or subscribe to content they are interested in.

**Comment by esethna** [comment]
Interesting, would the admin set the keywords that would automatically add the hashtags?

**Reply by chetanyakan** [reply]
I was thinking of using NLP to extract the context from messages.

But using plugins with pre-defined keywords would simplify things so much, haha.

**Reply by jespino** [reply]
I did that before! hahaha I was playing around with the idea of adding an action button in the same way that we add webhooks now, and one of the proof of concept implemented there was the "autotag" button, that does exactly that, analyze the post using `RAKE` and extract the tags for that, then edit the post and add the tags at the end of the message :)

here: https://github.com/jespino/matteractions (https://github.com/jespino/matteractions/blob/master/actions/autotag.py)
and here: https://github.com/jespino/mattermost-post-actions-plugin

Also I added another auto-reaction using deep-moji, that was pretty cool too.

Are the very old plugin/service that i built for that :)

**Reply by chetanyakan** [reply]
Woah!
I was thinking of adding something to post props to search for such posts rather than modifying the post message.
But this idea is good too!

**Reply by esethna** [reply]
Nice! I think having admin configurability for keywords and associated tags would be super key here, in addition to the autogenerate ones

**Reply by chetanyakan** [reply]
Yes. We could have multiple post props.
One for admin configured tags and other for AI generated.

**Comment by Willyfrog** [comment]
instead of relying on AI, maybe we could have some sort of shortcut to turn a word into a hashtag?

e.g.: post a message, in the message posted ctrl+click on a word and that edits the message automatically and turns the clicked word into a hashtag

**Reply by chetanyakan** [reply]
Wish plugins had the ability to override right-click context menu.
